### DISCUSSION TITLE
How to implement Unique Attribute in property/field

### DISCUSSION THREAD

**Discussion by harifenny** [serenity-is/Serenity#5408] [Help] [answered]
In any condition we have to unique for property in database

https://www.w3schools.com/sql/sql_unique.asp

How to implement in Serenity? I just use [Unique] attribute but can not work

Thanks & Regars, 
Hari Setyawan

**Comment by minhhungit** [comment, marked as answer]
It has UniqueConstraint attribute, put it on your row class, like this

```csharp
[UniqueConstraint(new[] { "SupplierId", "OemNumber", "StockNumber", "YardLocation" })] 
public sealed class InventoryStoreRow : Row, IIdRow, INameRow { 
  [DisplayName("Id"), Identity] 
  public Int32? Id { 
    get { return Fields.Id[this]; } 
    set { Fields.Id[this] = value; } 
  }
}
```

**Reply by harifenny** [reply]
Hi minhhungit, thanks for your reply. From your suggestion, I just modify with add

`[UniqueConstraint(new[] { "ItemCode"})]`

In ItemRow class, and create unique constraint in database with this script

`Alter Table Item Add CONSTRAINT ItemCode UNIQUE (ItemCode)`

But, I got an error like this

![uniqueConstraintError-2021-01-29](https://user-images.githubusercontent.com/77719430/106232570-4b39db80-622f-11eb-942f-2dff5bc6d78c.png)

How to fix it?

**Reply by harifenny** [reply]
I am also trying to modify with this link [Handling unique key constraints #3395](https://github.com/serenity-is/Serenity/issues/3395) but I got still an error

**Reply by minhhungit** [reply]
After adding the attribute, it will prevent user insert duplicate row. The error said that you row with ItemCode 001 is existed, so that is expected

**Reply by minhhungit** [reply]
You might need to check your database and remove duplicated row

**Reply by harifenny** [reply]
Finally, I just copy paste like in UserRepository.cs to implement my problem. 

Thanks a lot

**Comment by harifenny** [comment]
Finally, I copy paste like in UserRepository.cs to implement my problem. Thanks a lot
